cdap-ui/app/hydrator/services/create/actions/available-plugins-action-creator.js

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const getPluginToArtifactMap = (plugins = []) => {
+  let typeMap = {};
+  plugins.forEach( plugin => {
+    typeMap[plugin.name] = typeMap[plugin.name] || [];
+    typeMap[plugin.name].push(plugin);
+  });
+  return typeMap;
+};
+
+const getDefaultVersionForPlugin = (plugin = {}, defaultVersionMap = {}) => {
+  if([Object.keys(plugin), Object.keys(defaultVersionMap)].indexOf(0) !== -1) {
```

Comment:
Since `Object.keys` returns an array, not sure how it can be compared to 0 here. Did you mean `Object.keys(plugin).length` and `Object.keys(defaultVersionMap).length`? If so, I think it'd be more straightforward to do `if (!Object.keys(plugin).length || !Object.keys(defaultVersionMap).length)`